evaluations/ipfs.md

```diff
@@ -0,0 +1,41 @@
+An evaluation of the IPFS paper
+===============================
+
+2021-01-06 Mark Spanbroek
+
+https://ipfs.io/ipfs/QmR7GSQM93Cx5eAg6a6yRzNde1FQv7uL6X1o4k7zrJa3LX/ipfs.draft3.pdf
+
+Goal of this evaluation is to find things to adopt or avoid while designing
+Dagger. It is not meant to be a criticism of IPFS.
+
+#### Pros:
+
++ IPFS is designed by simplifying, evolving, and connecting proven techniques
+  (§3)
++ Consists of a stack of separately described sub-protocols (§3)
++ Uses Coral DSHT to favor data that is nearby, reducing latency of lookup
+  (§2.1.2)
++ Uses proof-of-work in S/Kademlia to discourage Sybil attacks (§2.1.3)
++ Favors self-describing values such as multihash (§3.1) and multiaddr (§3.2.1)
++ BitSwap protocol for exchanging blocks supports multiple strategies (§3.4.2)
++ Uses content addressing (§3.5)
++ The Merkle DAG is simple, yet allows constucting filesystems,
+  key-value stores, databases, messaging system, etc.. (§3.5)
+
+#### Cons:
+
+- Kademlia prefers long-lived nodes, which is not ideal for mobile enviroments
```

Comment:
This is a consequence of using libp2p, which doesn't support "packet switching", it uses streaming. This can and will be alleviated with the use of quic.

Reply:
Quic will probably help, but you'd still need to update the DHT when you change ip address.

Reply:
I don't think there is an efficient routing and discovery mechanism besides a DHT, so it's a bit of a necessary tradeoff. Also, you don't necessarily update the DHT, once you've changed networks all you're previous connections have died and you need to do another round of discovery at which point other nodes learn your new IP.

Reply:
or, you broadcast your new address to the nodes already present in your buckets.

Reply:
Yes, indeed. Kademlia is probably the DHT to go with, however there are [other DHTs](https://aircconline.com/ijcnc/V9N5/9517cnc04.pdf) that claim to handle churn better.